modules/precisoBidAdapter.js

```diff
@@ -37,17 +36,16 @@ export const spec = {
     return Boolean(bid.bidId && bid.params && bid.params.publisherId && precisoBid);
   },
   buildRequests: buildRequests(endpoint),
-  interpretResponse,
+  interpretResponse: buildBidResponse,
   onBidWon,
   getUserSyncs: (syncOptions, serverResponses, gdprConsent, uspConsent) => {
     const isSpec = syncOptions.spec;
-    if (!Object.is(isSpec, true)) {
+    if ((!Object.is(isSpec, true)) && (storage.localStorageIsEnabled())) {
       let syncId = storage.getCookie('_sharedid');
```

Comment:
this only works if the publishers sets up sharedId to use a cookie with this name; it also bypasses consent/activity control settings (granted it's an unlikely scenario, but it's possible that a publisher may decide to not make sharedId available to Preciso in particular).

You can get sharedId (when available) as `bidRequest.userId.pubcid`, or as an EID with source 'pubcid.org' in `userIdAsEids`.

Reply:
Hi @dgirardi 
we are passing the sharedId to sync and prevent the unwanted bidrequests.  we are passing this sharedId if it is available only, otherwise we send it as NA. even if the sharedid is there or not getuserSyncs will happen. if the sharedid is present, it will pass that otherwise it consider as NA.

Thank you!

Reply:
If publishers decide to set their sharedId to a different browser storage key or location, your bidder will not see that and provide bids. You can avoid those scenarios by looking in the locations mentioned above.

Reply:
Hi @ncolletti  Can  I get bidRequest.userId.pubcid in getUserSyncs() ?